### ISSUE TITLE
Request for a feature in AnnotatedService to allow HttpService response

### ISSUE THREAD

**Issue by chungonn** [line/armeria#4137]
It would useful to allow HttpService as response in an Annotated service apart from the current HttpResponse. It would allow a developer access of the HttpRequest and ServiceRequestContext apart from the injected parameters. It would also allow the developer ability to create different HttpService subclass for the various use case.

Below is a sample of the  of an AnnotatedService with a HttpService response in Scala 3 syntax

```scala
class AnnontatedService:

    //1. Current HttpResponse return type
    @Get("/text")
    def text: HttpResponse = HttpResponse.of("Hi")

    //2. HttpService return type 
    @Get("/text2") 
    def text2: HttpService = new HttpService{
       override def serve(ctx: ServiceRequestContext, req: HttpRequest): HttpResponse =
          HttpResponse.from(() => HttpResponse.of("Howdy"), ctx.blockingTaskExecutor())
    }

   //3. A pseudo code of an Action subclass of HttpService
    @Get("/action") 
    def hello: HttpService = Action { request => 
        HttpResponse("Hi") 
    }
```

**Comment by trustin** [comment]
I agree this is gonna be useful in certain situations. Are you interested in sending a PR for this? We can give you a few pointers if you're interested.

**Comment by chungonn** [comment]
Hi Trustin,

FYI, I have developed a couple of Scala traits that would allow us to have the equivalent AnnotatedHttpService without any code change in Armeria, however this solution is sub optimal, a better solution would be Java PR. 

I did attempted to work on a Java version (PR) but failed because my Java skills are very rusty as I have not done Java development for more than a decade. I am keen to work on the Java PR once I have polished up my Java and have some spare cycles as we are currently developing a web-container using Armeria. 

Please provide me the few pointers that allow me to get started, if you able to wait for my PR. Thanks in advance

**Comment by chungonn** [comment]
> I agree this is gonna be useful in certain situations. Are you interested in sending a PR for this? We can give you a few pointers if you're interested.

I am ready to work on this issue, can you give me the necessary pointers to work on? I trust I would working on the Java source. I would be using Intellij for my development environment with Java 17. I would be using Armeria with Scala 3 and Java 17.

Thanks in advance

**Comment by ikhoon** [comment]
You can start from reading the code around:
https://github.com/line/armeria/blob/94a1bf40ff41e1cfa9cd0c1a08f7bde14a7153c9/core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedService.java#L353-L376
We can add a new branch for `HttpService`.

**Comment by chungonn** [comment]
Noted with thanks!

**Comment by chungonn** [comment]
I studied the code in closer details and I  realised that the current code base is very generic and flexible. It basically provides the manipulation of all  types of input and output. 

Before doing a deep dive into the code base, I achieved what I need by creating an annotated RequestConverterFunction and ResponseConverterFunction classes corresponding. Now, I am able to achieve what I need without creating any annotated xxxConverterFunctions classes simply by invoking ServiceBuilder#annotatedServiceExtensions() method with the required converter function instances.

Thanks for providing such nice APIs and I am sorry for not spending enough time to study the available APIs before requesting for the feature.

I would close the request ticket.
